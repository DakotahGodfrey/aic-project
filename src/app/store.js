import { configureStore } from '@reduxjs/toolkit';
import homeReducer from 'app/slices/homeSlice';
import galleryReducer from 'app/slices/gallerySlice';
import artistsReducer from 'app/slices/artistsSlice';
import artworksReducer from 'app/slices/artworkSlice';
import artistReducer from 'app/slices/artistSlice';

export const store = configureStore({
  reducer: {
    home: homeReducer,
    gallery: galleryReducer,
    artists: artistsReducer,
    artwork: artworksReducer,
    artist: artistReducer,
  },
});
