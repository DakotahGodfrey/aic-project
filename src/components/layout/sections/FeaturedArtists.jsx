import { CarouselWrapper, CarouselCard } from 'components/layout/Layout';
import { Link } from 'react-router-dom';
import { formatImageUrl } from 'utils/helpers';
import placeholder from 'images/placeholder.svg';

const FeaturedArtists = ({ artists, coverImages }) => {
  return (
    <CarouselWrapper>
      {artists.map((artist, index) =>
        artist.id ? (
          <Link to={`/artists/${artist.id}`} key={artist.id}>
            <CarouselCard isArtist>
              <img
                src={formatImageUrl(coverImages[index].image_id)}
                alt={coverImages[index].thumbnail.alt_text ?? ''}
              />
              <figcaption>
                <p>{artist.title}</p>
              </figcaption>
            </CarouselCard>
          </Link>
        ) : (
          <CarouselCard isArtist key={index}>
            <img src={placeholder} alt='loading' />
            <figcaption>
              <p>Loading...</p>
            </figcaption>
          </CarouselCard>
        )
      )}
    </CarouselWrapper>
  );
};
export default FeaturedArtists;
