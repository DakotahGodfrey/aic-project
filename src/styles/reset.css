*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
}
html {
  font-size: 10px;
}
ol,
ul {
  list-style: none;
}
a {
  text-decoration: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
::-webkit-scrollbar {
  width: 0; /* Remove scrollbar space */
  background: transparent; /* Optional: just make scrollbar invisible */
}
