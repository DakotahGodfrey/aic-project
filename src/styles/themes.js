const defaultTheme = {
  textColor: '#222',
  secondaryColor: '#fff',
  navBG: '#111',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#666',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};
const darkTheme = {
  textColor: '#eee',
  navBG: '#111',
  secondaryColor: '#151515',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#ddd',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};

const getFromTheme = (propName) => (props) => props.theme[propName];
export { defaultTheme, darkTheme, getFromTheme };
