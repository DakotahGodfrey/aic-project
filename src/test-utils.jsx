import * as React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

function render(ui, { theme, store, ...options } = {}) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Router>{children}</Router>
      </ThemeProvider>
    </Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export * from '@testing-library/react';
// override React Testing Library's render with our own
export { render };
