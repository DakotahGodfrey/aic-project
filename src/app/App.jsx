import React, { useEffect, useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { darkTheme, defaultTheme } from 'styles/themes';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  useLocation,
} from 'react-router-dom';
import SiteNav from 'components/layout/navigation/SiteNav';
import { siteLinks } from 'utils/data';
import Home from 'components/screens/Home';
import Gallery from 'components/screens/Gallery';
import Artists from 'components/screens/Artists';
import Artwork from 'components/screens/Artwork';
import Artist from 'components/screens/Artist';
import AboutApp from 'components/screens/AboutApp';
import NotFound from 'components/screens/NotFound';
import ScrollUpButton from 'components/layout/buttons/ScrollUpButton';

function ScrollToTop() {
  const { pathname } = useLocation();

  useEffect(() => {
    if (window instanceof Window) {
      window.scrollTo(0, 0);
    }
  }, [pathname]);

  return null;
}

const App = () => {
  const [theme, setTheme] = useState('default');

  const handleThemeToggle = () => {
    if (theme === 'default') {
      setTheme('darkTheme');
    } else {
      setTheme('default');
    }
    console.log(theme);
  };
  return (
    <ThemeProvider theme={theme === 'default' ? defaultTheme : darkTheme}>
      <Router>
        <div
          id='app'
          style={
            theme === 'default'
              ? { backgroundColor: '#eee' }
              : { backgroundColor: '#111' }
          }
        >
          <ScrollToTop />
          <SiteNav
            links={siteLinks}
            toggleTheme={handleThemeToggle}
            theme={theme}
          />
          <Switch>
            <Route exact path={'/'} component={Home} />
            <Route exact path={'/gallery'} component={Gallery} />
            <Route exact path={'/artists'} component={Artists} />
            <Route exact path={'/gallery/artwork/:id'} component={Artwork} />
            <Route exact path={'/artists/:id'} component={Artist} />
            <Route exact path={'/about'} component={AboutApp} />
            <Route component={NotFound} />
          </Switch>
          <ScrollUpButton />
        </div>
      </Router>
    </ThemeProvider>
  );
};

export default App;
