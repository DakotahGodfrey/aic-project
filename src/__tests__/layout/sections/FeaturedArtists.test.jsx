import { store } from 'app/store';
import FeaturedArtists from 'components/layout/sections/FeaturedArtists';
import { defaultTheme } from 'styles/themes';
import { render, screen } from 'test-utils';
const artists = [
  { id: '1', title: 'artist 1' },
  { id: '2', title: 'artist 2' },
  { id: '3', title: 'artist 3' },
];
const coverImages = [
  { id: '1', image_id: '1', thumbnail: { alt_text: 'image1alt' } },
  { id: '2', image_id: '2', thumbnail: { alt_text: 'image2alt' } },
  { id: '3', image_id: '3', thumbnail: { alt_text: 'image3alt' } },
];
describe('<FeaturedArtists />', () => {
  it.skip('throws an error if no featured artists', () => {
    expect(() => {
      render(<FeaturedArtists />, { theme: defaultTheme, store: store });
    }).toThrow();
  });
  it.skip('throws an error if no cover images', () => {
    expect(() => {
      render(<FeaturedArtists artists={artists} />, {
        theme: defaultTheme,
        store: store,
      });
    }).toThrow();
  });
  it('matches the snapshot', () => {
    const wrapper = render(
      <FeaturedArtists artists={artists} coverImages={coverImages} />,
      { theme: defaultTheme, store: store }
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders a link to /artists/:id for each artist', () => {
    render(<FeaturedArtists artists={artists} coverImages={coverImages} />, {
      theme: defaultTheme,
      store: store,
    });
    for (let i = 0; i < artists.length; i++) {
      expect(screen.getAllByRole('link')[i]).toHaveAttribute(
        'href',
        `/artists/${artists[i].id}`
      );
    }
  });
  it('renders an img for each cover image with the correct alt_text', () => {
    render(<FeaturedArtists artists={artists} coverImages={coverImages} />, {
      theme: defaultTheme,
      store: store,
    });
    for (let i = 0; i < coverImages.length; i++) {
      expect(screen.getAllByRole('img')[i].src).toContain(
        coverImages[i].image_id
      );
      expect(screen.getAllByRole('img')[i].alt).toEqual(
        coverImages[i].thumbnail.alt_text
      );
    }
  });
  it('renders an element with the artists title', () => {
    render(<FeaturedArtists artists={artists} coverImages={coverImages} />, {
      theme: defaultTheme,
      store: store,
    });
    for (let i = 0; i < artists.length; i++) {
      expect(screen.getByText(artists[i].title)).toBeInTheDocument();
    }
  });
});
