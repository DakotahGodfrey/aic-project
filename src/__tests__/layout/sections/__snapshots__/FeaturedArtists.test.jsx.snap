// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedArtists /> matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlnjwi eaKyIO"
      >
        <a
          href="/artists/1"
        >
          <figure
            class="sc-hKFxyN dsCYYK"
          >
            <img
              alt="image1alt"
              src="https://www.artic.edu/iiif/2/1/full/843,/0/default.jpg"
            />
            <figcaption>
              <p>
                artist 1
              </p>
            </figcaption>
          </figure>
        </a>
        <a
          href="/artists/2"
        >
          <figure
            class="sc-hKFxyN dsCYYK"
          >
            <img
              alt="image2alt"
              src="https://www.artic.edu/iiif/2/2/full/843,/0/default.jpg"
            />
            <figcaption>
              <p>
                artist 2
              </p>
            </figcaption>
          </figure>
        </a>
        <a
          href="/artists/3"
        >
          <figure
            class="sc-hKFxyN dsCYYK"
          >
            <img
              alt="image3alt"
              src="https://www.artic.edu/iiif/2/3/full/843,/0/default.jpg"
            />
            <figcaption>
              <p>
                artist 3
              </p>
            </figcaption>
          </figure>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlnjwi eaKyIO"
    >
      <a
        href="/artists/1"
      >
        <figure
          class="sc-hKFxyN dsCYYK"
        >
          <img
            alt="image1alt"
            src="https://www.artic.edu/iiif/2/1/full/843,/0/default.jpg"
          />
          <figcaption>
            <p>
              artist 1
            </p>
          </figcaption>
        </figure>
      </a>
      <a
        href="/artists/2"
      >
        <figure
          class="sc-hKFxyN dsCYYK"
        >
          <img
            alt="image2alt"
            src="https://www.artic.edu/iiif/2/2/full/843,/0/default.jpg"
          />
          <figcaption>
            <p>
              artist 2
            </p>
          </figcaption>
        </figure>
      </a>
      <a
        href="/artists/3"
      >
        <figure
          class="sc-hKFxyN dsCYYK"
        >
          <img
            alt="image3alt"
            src="https://www.artic.edu/iiif/2/3/full/843,/0/default.jpg"
          />
          <figcaption>
            <p>
              artist 3
            </p>
          </figcaption>
        </figure>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
